lib/core/messageReceiver.ts

```diff
@@ -315,11 +348,11 @@ export class MessageReceiver extends LinkEntity {
       }
       try {
         await this._onMessage(bMessage);
-        clearTimerAndStopExecution();
+        this._clearMessageLockRenewTimer(bMessage.messageId as string);
```

Comment:
Should the `clearTimerAndStopExecution` method check to see if a settlement has been started on the message before doing this? Or we assume that once the user's message handler has been called, we can simply remove the renewal lock?

Reply:
Good question. For the streaming receiver, the customer's message handler is the right place for the customer to settle the message. The message can surely be cached by the customer outside the message handler and the customer may choose to dispose the message at a later stage. However, that would be really odd. 

Have posted this on Teams as well. My understanding of the expected behavior is that either the customer settles the message from within the message handler, or we settle it for her/him (if autoComplete is true).

Currently, the auto message lock renewal timer is cleared whenever
- the action to settle a message has started OR
- the customer's onMessage handler completes execution.